File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        int n=nums.length;
        for(int num:nums){
            mp.put(num,mp.getOrDefault(num,0)+1);
        }

        for(int i=0;i<n;++i){
            if(mp.get(nums[i])>n/2) return nums[i];
        }
        return -1;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int cand=0;
        int count=0;
        for(int &num: nums){
            if(count==0){
                cand=num;
            }
            if(cand==num){
                count++;
            }else{
                count--;
            }
        }
        return cand;
    }
};